### PR TITLE
test(client): Add cockroachdb blog tests

### PR DESCRIPTION
**Pull request by garrensmith**

### CODE REVIEW

File: packages/client/src/__tests__/integration/happy/blog-env-cockroach/schema.prisma
Comment (do4gr) [inline packages/client/src/__tests__/integration/happy/blog-env-cockroach/schema.prisma:2]: This looks like you're testing a cockroach connection string, but with a postgres provider? Shouldn't this have the cockroach provider and the preview flag?

Comment (garrensmith) [reply packages/client/src/__tests__/integration/happy/blog-env-cockroach/schema.prisma]: Fixed.

File: .github/workflows/test.yml
Comment (Jolg42) [inline .github/workflows/test.yml:238]:
```suggestion
```

I don't think we have yet these integration-tests (in https://github.com/prisma/prisma/tree/main/packages/integration-tests) so we would need to remove that for now

Comment (garrensmith) [reply .github/workflows/test.yml:238]: How do I get it to run https://github.com/prisma/prisma/blob/d267b1756cddd26b6ce4977601032b1331474443/packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/test.ts#L71

Comment (Jolg42) [reply .github/workflows/test.yml:238]: Naming might be confusing here, "integration-tests" is a different package and different tests, here `prisma/packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/test.ts` is part of client package and with be matched by the structure/name automatically and will be added to the tests run without doing anything else.

Comment (Jolg42) [reply .github/workflows/test.yml:238]: Ran here using binary engine and passed
https://github.com/prisma/prisma/runs/4580857834?check_suite_focus=true#step:9:812

Ran here using library engine and passed
https://github.com/prisma/prisma/runs/4580857643?check_suite_focus=true#step:9:829

File: .buildkite/test/docker-compose.14.yml
Comment (Jolg42) [inline .buildkite/test/docker-compose.14.yml:33]:
```suggestion
      - TEST_COCKROACH_URI=postgresql://prisma@cockroachdb:26257/
```

File: .buildkite/test/docker-compose.14.yml
Comment (Jolg42) [inline .buildkite/test/docker-compose.14.yml:32]:
```suggestion
      # CockroachDB
```

File: .buildkite/publish/docker-compose.yml
Comment (Jolg42) [inline .buildkite/publish/docker-compose.yml:33]:
```suggestion
      # CockroachDB
      - TEST_COCKROACH_URI=postgresql://prisma@cockroachdb:26257/
```

File: packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/test.ts
Comment (aqrln) [inline packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/test.ts:18]:
```suggestion
    let originalConnectionString = process.env.TEST_COCKROACH_URI || 'postgresql://prisma@127.0.0.1:26257/tests'
```

File: docker/docker-compose.yml
Comment (aqrln) [inline docker/docker-compose.yml:51]: Do we have an ARM64 build of this image? If not, it would be nice to add an override to `docker-compose.arm64.yml` with either an alternative image or just

```yaml
cockroachdb:
  profiles:
    - donotstart
```

and skip the test based on some condition.

Comment (aqrln) [reply docker/docker-compose.yml:51]: (not blocking this PR because of this, I can do that separately after it lands on `main`)

Comment (garrensmith) [reply docker/docker-compose.yml:51]: The image works on ARM64. I've been using it locally.

**Comment by Jolg42** [comment]
It looks like the tests in `packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/test.ts` are a mix of tests founds from other files and it's hard to understand what's actually covered. For example it looks like it's missing tests for `$executeRawUnsafe`

Where did you copy these tests from?

File: packages/client/src/runtime/getPrismaClient.ts
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:633]:
```suggestion
            throw new Error(`The ${this._activeProvider} provider does not support $executeRaw`)
```

File: packages/client/src/runtime/getPrismaClient.ts
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:783]:
```suggestion
            throw new Error(`The ${this._activeProvider} provider does not support $queryRaw`)
```

File: packages/client/src/runtime/getPrismaClient.ts
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:802]:
```suggestion
            throw new Error(`The ${this._activeProvider} provider does not support $queryRaw`)
```

File: packages/client/src/runtime/getPrismaClient.ts
Comment (Jolg42) [inline packages/client/src/runtime/getPrismaClient.ts:614]:
```suggestion
            throw new Error(`The ${this._activeProvider} provider does not support $executeRaw`)
```

**Comment by garrensmith** [comment]
I've taken the tests from the postgres one and a little bit from the MySQL tests.

File: TESTING.md
Comment (millsp) [inline TESTING.md:31]:
```suggestion
export TEST_COCKROACH_URI=postgresql://prisma@localhost:26257/
```

File: docker/docker-compose.yml
Comment (millsp) [inline docker/docker-compose.yml:17]:
```suggestion
  # - TEST_COCKROACH_URI=postgresql://prisma@localhost:26257/
```

File: .buildkite/test/docker-compose.14.yml
Comment (Jolg42) [inline .buildkite/test/docker-compose.14.yml:53]: Nice catch

**Review by millsp — CHANGES_REQUESTED** [review]
Nice work, and congrats on your first TS PR! :tada:

**Review by Jolg42 — APPROVED** [review]
Well done! 💚

**Comment by garrensmith** [comment]
Thanks for all the help. I have this test for executeRawUnsafe https://github.com/prisma/prisma/pull/10772/files#diff-f333cde87774f1eb57d0090390e4719aa0a88a74ec3cd8165523e68232875340R314

**Comment by Jolg42** [comment]
@garrensmith indeed, I didn't see it, looks good to merge!

**Comment by aqrln** [comment]
Let's merge this PR first, I'll rebase my branch and make it work with https://github.com/prisma/prisma/pull/10526

File: docker/docker-compose.yml
Comment (aqrln) [inline docker/docker-compose.yml:55]: Should we also attach a volume to persist data similar to other databases here? LGTM either way though.

**Review by aqrln — APPROVED** [review]

File: docker/docker-compose.yml
Comment (garrensmith) [reply docker/docker-compose.yml:55]: What do you persist the data for?

Comment (janpio) [reply docker/docker-compose.yml:55]: (Might just be a think we have been cargo culting with no actual use...)

File: packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma
Comment (janpio) [inline packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma:15]: Space inside the triple comment intended?

File: packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma
Comment (janpio) [inline packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma:8]:
```suggestion
  previewFeatures = ["cockroachdb"]
```
unless intentional

File: packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma
Comment (aqrln) [reply packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma:15]: I think it was copypasted from the other tests. We have a ton of tests with the same typo (maybe there used to be a bug in `prisma-fmt` in the past?).

Comment (millsp) [reply packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma:15]: That comes from our other tests: 171 occurrences in 90 files. So something to fix but also mostly not needed as we only test comments in our type gen snapshots.

File: docker/docker-compose.yml
Comment (aqrln) [reply docker/docker-compose.yml:55]: Yeah, on the second thought I would rather prefer we don't persist data in the other services too instead.

File: packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma
Comment (Jolg42) [reply packages/client/src/__tests__/integration/happy/blog-env-cockroachdb/schema.prisma:8]: Casing doesn't matter for previewFeatures though

**Review by millsp — APPROVED** [review]

**Pull request merged**
